File: tarotScoreUi/src/app/dtos/round-dto.ts

import {PlayerDto} from './player-dto';

export type ContractType = 'PETITE' | 'GARDE' | 'GARDE_SANS' | 'GARDE_CONTRE';
export type PetitAuBoutStatus = 'NON' | 'GAGNE' | 'PERDU';
export type ChelemStatus = 'NON' | 'ANNONCE_REUSSI' | 'ANNONCE_RATE' | 'REUSSI_NON_ANNONCE';

export interface RoundDto {
  id?: string;

  contractType: ContractType;
  petitStatus: PetitAuBoutStatus;
  createdDate: Date;
  chelemStatus: ChelemStatus;

  scoreAttaque: number;
  nombreBouts: 0 | 1 | 2 | 3;

  taker: PlayerDto;
  called?: PlayerDto;
  gameId: string;

  players: PlayerDto[];

  poigneeSimples: PlayerDto[];
  poigneeDoubles: PlayerDto[];
  poigneeTriple: PlayerDto[];

  misereSimples: PlayerDto[];
  misereDoubles: PlayerDto[];
}

File: tarotScoreApi/src/main/java/fr/ladevie/tarot/score/dtos/RoundDTO.java
package fr.ladevie.tarot.score.dtos;

import fr.ladevie.tarot.score.business.enums.ChelemStatus;
import fr.ladevie.tarot.score.business.enums.ContractType;
import fr.ladevie.tarot.score.business.enums.PetitAuBoutStatus;

import java.util.List;
import java.util.UUID;

public class RoundDTO {

    private UUID id;

    private ContractType contractType;
    private PetitAuBoutStatus petitStatus;
    private ChelemStatus chelemStatus;

    private Integer scoreAttaque;

    private UserDTO taker;
    private UserDTO called;
    private UUID gameId;

    private List<UserDTO> players;

    private List<UserDTO> poigneeSimples;
    private List<UserDTO> poigneeDoubles;
    private List<UserDTO> poigneeTriple;
    private List<UserDTO> misereSimples;
    private List<UserDTO> misereDoubles;
}